### PR TITLE
Support connect redis with username and password

### PR DESCRIPTION
**Pull request by HaitaoDeng**
image
image
image

### CODE REVIEW

File: test/api/redis/redis_api_set_get_test.dart [hunk 1]
```diff
@@ -1,10 +1,9 @@
 import 'package:flutter_test/flutter_test.dart';
-import 'package:paas_dashboard_flutter/api/redis/redis_api.dart';
 
 void main() {
   test("test_set_get", () async {
-    await RedisApi.set("localhost", 6379, "", "", "key", "value");
-    var val = await RedisApi.get("localhost", 6379, "", "", "key");
-    print(val);
+    // await RedisApi.set("localhost", 6379, "", "", "key", "value");
+    // var val = await RedisApi.get("localhost", 6379, "", "", "key");
```
Comment (Shoothzj) [inline test/api/redis/redis_api_set_get_test.dart:6]: why change this

File: lib/api/redis/redis_api.dart [hunk 2]
```diff
@@ -20,15 +20,78 @@
...
+
+  static Future<dynamic> delete(String host, int port, String password, String key) async {
+    dynamic rs;
+    final command = await getCommand(host, port, password);
+    rs = await command.send_object(["DEL", key]).onError((error, stackTrace) => throw Exception('redis error: $error'));
+    return rs;
+  }
+
+  static dynamic keys(String host, int port, String password, String patten) async {
+    dynamic rs;
```
Comment (Shoothzj) [inline lib/api/redis/redis_api.dart:47]: put in one line

File: pubspec.lock [hunk 3]
```diff
@@ -380,128 +380,135 @@ packages:
...
       name: watcher
-      url: "https://pub.dartlang.org"
+      url: "https://pub.flutter-io.cn"
     source: hosted
     version: "1.0.1"
+  win32:
+    dependency: transitive
+    description:
+      name: win32
+      url: "https://pub.flutter-io.cn"
```
Comment (Shoothzj) [inline pubspec.lock:509]: please commit use `https://pub.dartlang.org`

Comment (HaitaoDeng) [reply pubspec.lock:509]: auto modify every time~

File: lib/api/redis/redis_api.dart (hunk 2 above)
Comment (HaitaoDeng) [reply lib/api/redis/redis_api.dart:47]: OK

**Review by goflutterjava — APPROVED** [review]

**Pull request merged**